### ISSUE TITLE
TLS not working

### ISSUE THREAD

**Issue by aug70** [traefik/traefik#4227]
### Do you want to request a *feature* or report a *bug*?

Bug

### What did you do?

Deploy with tls config

### What did you expect to see?

Normal operation with http redirecting to https

### What did you see instead?

```time="2018-11-24T04:57:18Z" level=error msg="Error configuring TLS for ingress kube-system/traefik-web-ui: secret kube-system/traefik-ui-tls-cert does not exist"```

### Output of `traefik version`: (_What version of Traefik are you using?_)

```
time="2018-11-24T04:57:18Z" level=info msg="Traefik version v1.7.4 built on 2018-10-30_10:44:30AM"
```

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

```yml
apiVersion: v1
kind: ConfigMap
metadata:
  name: traefik-conf
data:
  traefik.toml: |
    # traefik.toml
    logLevel = "INFO"
    defaultEntryPoints = ["http", "https"]
    [entryPoints]
      [entryPoints.http]
      address = ":80"
        [entryPoints.http.redirect]
        entryPoint = "https"
      [entryPoints.https]
      address = ":443"
        [entryPoints.https.tls]
          [[entryPoints.https.tls.certificates]]
          certFile = "/ssl/tls.crt"
          keyFile = "/ssl/tls.key"
    [kubernetes]

```

### If applicable, please paste the log output in DEBUG level (`--logLevel=DEBUG` switch)

```shell
time="2018-11-24T04:57:18Z" level=info msg="Using TOML configuration file /config/traefik.toml"
time="2018-11-24T04:57:18Z" level=info msg="No tls.defaultCertificate given for https: using the first item in tls.certificates as a fallback."
time="2018-11-24T04:57:18Z" level=info msg="Traefik version v1.7.4 built on 2018-10-30_10:44:30AM"
time="2018-11-24T04:57:18Z" level=info msg="\nStats collection is disabled.\nHelp us improve Traefik by turning this feature on :)\nMore details on: https://docs.traefik.io/basics/#collected-data\n"
time="2018-11-24T04:57:18Z" level=info msg="Preparing server http &amp;{Address::80 TLS:&lt;nil&gt; Redirect:0xc0003e89c0 Auth:&lt;nil&gt; WhitelistSourceRange:[] WhiteList:&lt;nil&gt; Compress:false ProxyProtocol:&lt;nil&gt; ForwardedHeaders:0xc000347c40} with readTimeout=0s writeTimeout=0s idleTimeout=3m0s"
time="2018-11-24T04:57:18Z" level=info msg="Preparing server https &amp;{Address::443 TLS:0xc000490870 Redirect:&lt;nil&gt; Auth:&lt;nil&gt; WhitelistSourceRange:[] WhiteList:&lt;nil&gt; Compress:false ProxyProtocol:&lt;nil&gt; ForwardedHeaders:0xc000347c60} with readTimeout=0s writeTimeout=0s idleTimeout=3m0s"
time="2018-11-24T04:57:18Z" level=info msg="Starting server on :80"
time="2018-11-24T04:57:18Z" level=info msg="Starting provider configuration.ProviderAggregator {}"
time="2018-11-24T04:57:18Z" level=info msg="Starting server on :443"
time="2018-11-24T04:57:18Z" level=info msg="Starting provider *kubernetes.Provider {\"Watch\":true,\"Filename\":\"\",\"Constraints\":[],\"Trace\":false,\"TemplateVersion\":0,\"DebugLogGeneratedTemplate\":false,\"Endpoint\":\"\",\"Token\":\"\",\"CertAuthFilePath\":\"\",\"DisablePassHostHeaders\":false,\"EnablePassTLSCert\":false,\"Namespaces\":null,\"LabelSelector\":\"\",\"IngressClass\":\"\",\"IngressEndpoint\":null}"
time="2018-11-24T04:57:18Z" level=info msg="ingress label selector is: \"\""
time="2018-11-24T04:57:18Z" level=info msg="Creating in-cluster Provider client"
time="2018-11-24T04:57:18Z" level=error msg="Error configuring TLS for ingress kube-system/traefik-web-ui: secret kube-system/traefik-ui-tls-cert does not exist"
time="2018-11-24T04:57:18Z" level=info msg="Server configuration reloaded on :443"
time="2018-11-24T04:57:18Z" level=info msg="Server configuration reloaded on :80"
time="2018-11-24T04:57:18Z" level=warning msg="Endpoints not available for kube-system/traefik-web-ui"
time="2018-11-24T04:57:18Z" level=warning msg="Endpoints not available for kube-system/traefik-web-ui"
time="2018-11-24T04:57:20Z" level=info msg="Server configuration reloaded on :80"
time="2018-11-24T04:57:20Z" level=info msg="Server configuration reloaded on :443"

```

When I check the secret it's there...
```
kubectl get secret -n kube-system traefik-ui-tls-cert
NAME                  TYPE                DATA   AGE
traefik-ui-tls-cert   kubernetes.io/tls   2      20m
```

This has been mentioned in another ticket and closed without much investigation. (I'll reference the ticket here if I can find it)

**Comment by athoune** [comment]
I read `secret kube-system/traefik-ui-tls-cert does not exist` in your example.

**Comment by aug70** [comment]
@athoune yes. Are you asking a question? If you read the whole post, the secret exists -- proof at the bottom of the post. It's just Traefik doesn't see it or can't find it.

**Comment by dtomcej** [comment]
@aug70 Does the certificate actually work? or is it just the error message you are encountering?

There is a known issue: (https://github.com/containous/traefik/issues/3667) That references that error message. 

If you are finding that the certificate is _not_ being loaded at all by traefik, I would check the RBAC that your traefik pod serviceAccount runs with. It is possible that traefik may not have the correct permissions to access those secrets, even if the serviceAccount you use to run `kubectl` does.

**Comment by aug70** [comment]
@dtomcej The certificate is legit and I use it in many other places but it's a wild card certificate 🤔

* I just diffed my traefik-rbac.yaml file against https://github.com/containous/traefik/blob/master/examples/k8s/traefik-rbac.yaml they are same.
* ui.yaml file same as this https://github.com/containous/traefik/blob/master/examples/k8s/ui.yaml plus this in the end.
```
  tls:
   - secretName: traefik-ui-tls-cert
```
* Here is how I create the secret:
```
kubectl create secret -n kube-system generic traefik-ui-tls-cert \
    --from-file=tls.key=<(cat myprivkey.pem) \
    --from-file=tls.crt=<(cat myfullchain.pem) \
    -oyaml --dry-run | \
    cat <(cat -) <(echo "type: kubernetes.io/tls") | \
    kubectl create -f -
```
* Also checked on minikube, same error...

Works just fine without https

**Comment by dtomcej** [comment]
@aug70 Can you try enabling the debug logs? I want to see the events provided by k8s.

Also of note: Is there a particular reason you are patching the secret type? I don't _think_ it should matter, but can you try without patching the type, and leaving it `Opaque`?

Thanks!

**Comment by aug70** [comment]
```
time="2018-12-08T05:56:53Z" level=info msg="Using TOML configuration file /config/traefik.toml"
time="2018-12-08T05:56:53Z" level=info msg="No tls.defaultCertificate given for https: using the first item in tls.certificates as a fallback."
time="2018-12-08T05:56:53Z" level=info msg="Traefik version v1.7.5 built on 2018-12-03_11:01:00AM"
time="2018-12-08T05:56:53Z" level=debug msg="Global configuration loaded {\"LifeCycle\":{\"RequestAcceptGraceTimeout\":0,\"GraceTimeOut\":10000000000},\"GraceTimeOut\":0,\"Debug\":false,\"CheckNewVersion\":true,\"SendAnonymousUsage\":false,\"AccessLogsFile\":\"\",\"AccessLog\":null,\"TraefikLogsFile\":\"\",\"TraefikLog\":null,\"Tracing\":null,\"LogLevel\":\"DEBUG\",\"EntryPoints\":{\"http\":{\"Address\":\":80\",\"TLS\":null,\"Redirect\":{\"entryPoint\":\"https\"},\"Auth\":null,\"WhitelistSourceRange\":null,\"WhiteList\":null,\"Compress\":false,\"ProxyProtocol\":null,\"ForwardedHeaders\":{\"Insecure\":true,\"TrustedIPs\":null}},\"https\":{\"Address\":\":443\",\"TLS\":{\"MinVersion\":\"\",\"CipherSuites\":null,\"Certificates\":[{\"CertFile\":\"/ssl/tls.crt\",\"KeyFile\":\"/ssl/tls.key\"}],\"ClientCAFiles\":null,\"ClientCA\":{\"Files\":null,\"Optional\":false},\"DefaultCertificate\":{\"CertFile\":\"/ssl/tls.crt\",\"KeyFile\":\"/ssl/tls.key\"},\"SniStrict\":false},\"Redirect\":null,\"Auth\":null,\"WhitelistSourceRange\":null,\"WhiteList\":null,\"Compress\":false,\"ProxyProtocol\":null,\"ForwardedHeaders\":{\"Insecure\":true,\"TrustedIPs\":null}}},\"Cluster\":null,\"Constraints\":[],\"ACME\":null,\"DefaultEntryPoints\":[\"http\",\"https\"],\"ProvidersThrottleDuration\":2000000000,\"MaxIdleConnsPerHost\":200,\"IdleTimeout\":0,\"InsecureSkipVerify\":false,\"RootCAs\":null,\"Retry\":null,\"HealthCheck\":{\"Interval\":30000000000},\"RespondingTimeouts\":null,\"ForwardingTimeouts\":null,\"AllowMinWeightZero\":false,\"KeepTrailingSlash\":false,\"Web\":null,\"Docker\":null,\"File\":null,\"Marathon\":null,\"Consul\":null,\"ConsulCatalog\":null,\"Etcd\":null,\"Zookeeper\":null,\"Boltdb\":null,\"Kubernetes\":{\"Watch\":true,\"Filename\":\"\",\"Constraints\":[],\"Trace\":false,\"TemplateVersion\":0,\"DebugLogGeneratedTemplate\":false,\"Endpoint\":\"\",\"Token\":\"\",\"CertAuthFilePath\":\"\",\"DisablePassHostHeaders\":false,\"EnablePassTLSCert\":false,\"Namespaces\":null,\"LabelSelector\":\"\",\"IngressClass\":\"\",\"IngressEndpoint\":null},\"Mesos\":null,\"Eureka\":null,\"ECS\":null,\"Rancher\":null,\"DynamoDB\":null,\"ServiceFabric\":null,\"Rest\":null,\"API\":null,\"Metrics\":null,\"Ping\":null,\"HostResolver\":null}"
time="2018-12-08T05:56:53Z" level=info msg="\nStats collection is disabled.\nHelp us improve Traefik by turning this feature on :)\nMore details on: https://docs.traefik.io/basics/#collected-data\n"
time="2018-12-08T05:56:53Z" level=debug msg="Add certificate for domains *.my-domain.com"
time="2018-12-08T05:56:53Z" level=debug msg="Creating entry point redirect http -> https"
time="2018-12-08T05:56:53Z" level=info msg="Preparing server http &{Address::80 TLS:<nil> Redirect:0xc0000e5fc0 Auth:<nil> WhitelistSourceRange:[] WhiteList:<nil> Compress:false ProxyProtocol:<nil> ForwardedHeaders:0xc000418480} with readTimeout=0s writeTimeout=0s idleTimeout=3m0s"
time="2018-12-08T05:56:53Z" level=info msg="Preparing server https &{Address::443 TLS:0xc0002ed050 Redirect:<nil> Auth:<nil> WhitelistSourceRange:[] WhiteList:<nil> Compress:false ProxyProtocol:<nil> ForwardedHeaders:0xc000418520} with readTimeout=0s writeTimeout=0s idleTimeout=3m0s"
time="2018-12-08T05:56:53Z" level=info msg="Starting server on :80"
time="2018-12-08T05:56:53Z" level=debug msg="Add certificate for domains *.my-domain.com"
time="2018-12-08T05:56:53Z" level=info msg="Starting server on :443"
time="2018-12-08T05:56:53Z" level=info msg="Starting provider configuration.ProviderAggregator {}"
time="2018-12-08T05:56:53Z" level=info msg="Starting provider *kubernetes.Provider {\"Watch\":true,\"Filename\":\"\",\"Constraints\":[],\"Trace\":false,\"TemplateVersion\":0,\"DebugLogGeneratedTemplate\":false,\"Endpoint\":\"\",\"Token\":\"\",\"CertAuthFilePath\":\"\",\"DisablePassHostHeaders\":false,\"EnablePassTLSCert\":false,\"Namespaces\":null,\"LabelSelector\":\"\",\"IngressClass\":\"\",\"IngressEndpoint\":null}"
time="2018-12-08T05:56:53Z" level=debug msg="Using Ingress label selector: \"\""
time="2018-12-08T05:56:53Z" level=info msg="ingress label selector is: \"\""
time="2018-12-08T05:56:53Z" level=info msg="Creating in-cluster Provider client"
time="2018-12-08T05:56:53Z" level=debug msg="Received Kubernetes event kind *v1beta1.Ingress"
time="2018-12-08T05:56:53Z" level=error msg="Error configuring TLS for ingress kube-system/traefik-web-ui: secret kube-system/traefik-ui-tls-cert does not exist"
time="2018-12-08T05:56:53Z" level=debug msg="Configuration received from provider kubernetes: {}"
time="2018-12-08T05:56:53Z" level=debug msg="Add certificate for domains *.my-domain.com"
time="2018-12-08T05:56:53Z" level=info msg="Server configuration reloaded on :80"
time="2018-12-08T05:56:53Z" level=info msg="Server configuration reloaded on :443"
time="2018-12-08T05:56:53Z" level=debug msg="Received Kubernetes event kind *v1.Secret"
time="2018-12-08T05:56:53Z" level=warning msg="Endpoints not available for kube-system/traefik-web-ui"
time="2018-12-08T05:56:53Z" level=debug msg="Received Kubernetes event kind *v1.Secret"
time="2018-12-08T05:56:53Z" level=warning msg="Endpoints not available for kube-system/traefik-web-ui"
time="2018-12-08T05:56:53Z" level=debug msg="Skipping Kubernetes event kind *v1.Secret"
time="2018-12-08T05:56:53Z" level=debug msg="Configuration received from provider kubernetes: {\"backends\":{\"traefik-my-machine.my-domain.com/\":{\"loadBalancer\":{\"method\":\"wrr\"}}},\"frontends\":{\"traefik-my-machine.my-domain.com/\":{\"entryPoints\":[\"http\",\"https\"],\"backend\":\"traefik-my-machine.my-domain.com/\",\"routes\":{\"/\":{\"rule\":\"PathPrefix:/\"},\"traefik-my-machine.my-domain.com\":{\"rule\":\"Host:traefik-my-machine.my-domain.com\"}},\"passHostHeader\":true,\"priority\":0,\"basicAuth\":null}}}"
time="2018-12-08T05:56:53Z" level=debug msg="Received Kubernetes event kind *v1.Endpoints"
time="2018-12-08T05:56:53Z" level=debug msg="Received Kubernetes event kind *v1.Endpoints"
time="2018-12-08T05:56:53Z" level=debug msg="Skipping Kubernetes event kind *v1.Endpoints"
time="2018-12-08T05:56:53Z" level=debug msg="Configuration received from provider kubernetes: {\"backends\":{\"traefik-my-machine.my-domain.com/\":{\"servers\":{\"traefik-ingress-controller-rt5kn\":{\"url\":\"http://10.1.83.5:8080\",\"weight\":1}},\"loadBalancer\":{\"method\":\"wrr\"}}},\"frontends\":{\"traefik-my-machine.my-domain.com/\":{\"entryPoints\":[\"http\",\"https\"],\"backend\":\"traefik-my-machine.my-domain.com/\",\"routes\":{\"/\":{\"rule\":\"PathPrefix:/\"},\"traefik-my-machine.my-domain.com\":{\"rule\":\"Host:traefik-my-machine.my-domain.com\"}},\"passHostHeader\":true,\"priority\":0,\"basicAuth\":null}}}"
time="2018-12-08T05:56:54Z" level=debug msg="Received Kubernetes event kind *v1.Endpoints"
time="2018-12-08T05:56:54Z" level=debug msg="Received Kubernetes event kind *v1.Endpoints"
time="2018-12-08T05:56:54Z" level=debug msg="Skipping Kubernetes event kind *v1.Endpoints"
time="2018-12-08T05:56:54Z" level=debug msg="Configuration received from provider kubernetes: {\"backends\":{\"traefik-my-machine.my-domain.com/\":{\"servers\":{\"traefik-ingress-controller-rt5kn\":{\"url\":\"http://10.1.83.5:8080\",\"weight\":1},\"traefik-ingress-controller-tv5lg\":{\"url\":\"http://10.1.77.16:8080\",\"weight\":1}},\"loadBalancer\":{\"method\":\"wrr\"}}},\"frontends\":{\"traefik-my-machine.my-domain.com/\":{\"entryPoints\":[\"http\",\"https\"],\"backend\":\"traefik-my-machine.my-domain.com/\",\"routes\":{\"/\":{\"rule\":\"PathPrefix:/\"},\"traefik-my-machine.my-domain.com\":{\"rule\":\"Host:traefik-my-machine.my-domain.com\"}},\"passHostHeader\":true,\"priority\":0,\"basicAuth\":null}}}"
time="2018-12-08T05:56:54Z" level=debug msg="Received Kubernetes event kind *v1.Endpoints"
time="2018-12-08T05:56:54Z" level=debug msg="Received Kubernetes event kind *v1.Endpoints"
time="2018-12-08T05:56:54Z" level=debug msg="Skipping Kubernetes event kind *v1.Endpoints"
time="2018-12-08T05:56:54Z" level=debug msg="Configuration received from provider kubernetes: {\"backends\":{\"traefik-my-machine.my-domain.com/\":{\"servers\":{\"traefik-ingress-controller-bgc5p\":{\"url\":\"http://10.1.31.4:8080\",\"weight\":1},\"traefik-ingress-controller-rt5kn\":{\"url\":\"http://10.1.83.5:8080\",\"weight\":1},\"traefik-ingress-controller-tv5lg\":{\"url\":\"http://10.1.77.16:8080\",\"weight\":1}},\"loadBalancer\":{\"method\":\"wrr\"}}},\"frontends\":{\"traefik-my-machine.my-domain.com/\":{\"entryPoints\":[\"http\",\"https\"],\"backend\":\"traefik-my-machine.my-domain.com/\",\"routes\":{\"/\":{\"rule\":\"PathPrefix:/\"},\"traefik-my-machine.my-domain.com\":{\"rule\":\"Host:traefik-my-machine.my-domain.com\"}},\"passHostHeader\":true,\"priority\":0,\"basicAuth\":null}}}"
time="2018-12-08T05:56:55Z" level=debug msg="Received Kubernetes event kind *v1.Endpoints"
time="2018-12-08T05:56:55Z" level=debug msg="Skipping Kubernetes event kind *v1.Endpoints"
time="2018-12-08T05:56:55Z" level=debug msg="Received Kubernetes event kind *v1.Endpoints"
time="2018-12-08T05:56:55Z" level=debug msg="Skipping Kubernetes event kind *v1.Endpoints"
time="2018-12-08T05:56:55Z" level=debug msg="Add certificate for domains *.my-domain.com"
time="2018-12-08T05:56:55Z" level=debug msg="Wiring frontend traefik-my-machine.my-domain.com/ to entryPoint http"
time="2018-12-08T05:56:55Z" level=debug msg="Creating backend traefik-my-machine.my-domain.com/"
time="2018-12-08T05:56:55Z" level=debug msg="Adding TLSClientHeaders middleware for frontend traefik-my-machine.my-domain.com/"
time="2018-12-08T05:56:55Z" level=debug msg="Creating load-balancer wrr"
time="2018-12-08T05:56:55Z" level=debug msg="Creating server traefik-ingress-controller-bgc5p at http://10.1.31.4:8080 with weight 1"
time="2018-12-08T05:56:55Z" level=debug msg="Creating server traefik-ingress-controller-rt5kn at http://10.1.83.5:8080 with weight 1"
time="2018-12-08T05:56:55Z" level=debug msg="Creating server traefik-ingress-controller-tv5lg at http://10.1.77.16:8080 with weight 1"
time="2018-12-08T05:56:55Z" level=debug msg="Creating route / PathPrefix:/"
time="2018-12-08T05:56:55Z" level=debug msg="Creating route traefik-my-machine.my-domain.com Host:traefik-my-machine.my-domain.com"
time="2018-12-08T05:56:55Z" level=debug msg="Wiring frontend traefik-my-machine.my-domain.com/ to entryPoint https"
time="2018-12-08T05:56:55Z" level=debug msg="Creating backend traefik-my-machine.my-domain.com/"
time="2018-12-08T05:56:55Z" level=debug msg="Adding TLSClientHeaders middleware for frontend traefik-my-machine.my-domain.com/"
time="2018-12-08T05:56:55Z" level=debug msg="Creating load-balancer wrr"
time="2018-12-08T05:56:55Z" level=debug msg="Creating server traefik-ingress-controller-rt5kn at http://10.1.83.5:8080 with weight 1"
time="2018-12-08T05:56:55Z" level=debug msg="Creating server traefik-ingress-controller-tv5lg at http://10.1.77.16:8080 with weight 1"
time="2018-12-08T05:56:55Z" level=debug msg="Creating server traefik-ingress-controller-bgc5p at http://10.1.31.4:8080 with weight 1"
time="2018-12-08T05:56:55Z" level=debug msg="Creating route / PathPrefix:/"
time="2018-12-08T05:56:55Z" level=debug msg="Creating route traefik-my-machine.my-domain.com Host:traefik-my-machine.my-domain.com"
time="2018-12-08T05:56:55Z" level=debug msg="No entryPoint is defined to add the certificate MIIGCzCCBPOgAwIBAgISA0CUiZQjCkQ0GhkaYpjFej8sMA0GCS, it will be added to the default entryPoints: http, https"
time="2018-12-08T05:56:55Z" level=debug msg="Add certificate for domains *.my-domain.com"
time="2018-12-08T05:56:55Z" level=debug msg="Add certificate for domains *.my-domain.com"
time="2018-12-08T05:56:55Z" level=info msg="Server configuration reloaded on :443"
time="2018-12-08T05:56:55Z" level=debug msg="Certificates not added to non-TLS entryPoint http."
time="2018-12-08T05:56:55Z" level=info msg="Server configuration reloaded on :80"
time="2018-12-08T05:56:57Z" level=debug msg="Received Kubernetes event kind *v1.Endpoints"
time="2018-12-08T05:56:57Z" level=debug msg="Skipping Kubernetes event kind *v1.Endpoints"
time="2018-12-08T05:56:57Z" level=debug msg="Received Kubernetes event kind *v1.Endpoints"
```

**Comment by aug70** [comment]
More logs...
[traefik-ingress.log](https://github.com/containous/traefik/files/2691427/traefik-ingress.log)

**Comment by dtomcej** [comment]
I am reviewing these logs, and it appears that the certificate is found, and applied. The certificate is loaded asynchronously with the ingress, so there is a period of time that the secret will log as "not found" before the k8s api provides it. This issue is logged in: #3667.

```
The issue that you appear to be reporting is that the http -> https redirect does not work.

Can you provide us with a `curl -v

**Comment by aug70** [comment]
@dtomcej 
> The issue that you appear to be reporting is that the http -> https redirect does not work.

If your above statement is correct, then the following is a very confusing log message and must be corrected too... ```msg="Error configuring TLS for ingress kube-system/traefik-web-ui-ingress: secret kube-system/traefik-ui-tls-cert does not exist"```

**Comment by aug70** [comment]
@dtomcej I'm not sure what are you waiting from me? I have added all the log files that I could possibly add to this issue. This is not a redirect problem! It's a problem with TLS (https endpoint). You should focus on this error message:

> time="2018-11-24T04:57:18Z" level=error msg="Error configuring TLS for ingress kube-system/traefik-web-ui: secret kube-system/traefik-ui-tls-cert does not exist"

Once this is resolved, there won't be any problems (neither redirect nor 502) This seems like a primal functionality for traefik on kubernetes. I'm surprised that it doesn't work and also surprised that you reduced priority to P2.

**Comment by dtomcej** [comment]
Hello @aug70 .

From your results:

```
curl -v https://my.server.com/
*   Trying 10.225.192.55...
* TCP_NODELAY set
* Connected to my.server.com (10.225.192.55) port 443 (#0)
* ALPN, offering h2
* ALPN, offering http/1.1
* successfully set certificate verify locations:
*   CAfile: /etc/ssl/certs/ca-certificates.crt
  CApath: /etc/ssl/certs
* TLSv1.2 (OUT), TLS handshake, Client hello (1):
* TLSv1.2 (IN), TLS handshake, Server hello (2):
* TLSv1.2 (IN), TLS handshake, Certificate (11):
* TLSv1.2 (IN), TLS handshake, Server key exchange (12):
* TLSv1.2 (IN), TLS handshake, Server finished (14):
* TLSv1.2 (OUT), TLS handshake, Client key exchange (16):
* TLSv1.2 (OUT), TLS change cipher, Client hello (1):
* TLSv1.2 (OUT), TLS handshake, Finished (20):
* TLSv1.2 (IN), TLS handshake, Finished (20):
* SSL connection using TLSv1.2 / ECDHE-RSA-AES128-GCM-SHA256
* ALPN, server accepted to use h2
* Server certificate:
*  subject: CN=*.server.com
*  start date: Dec  8 03:46:35 2018 GMT
*  expire date: Mar  8 03:46:35 2019 GMT
*  subjectAltName: host "my.server.com" matched cert's "*.server.com"
*  issuer: C=US; O=Let's Encrypt; CN=Let's Encrypt Authority X3
*  SSL certificate verify ok.
* Using HTTP2, server supports multi-use
* Connection state changed (HTTP/2 confirmed)
* Copying HTTP/2 data in stream buffer to connection buffer after upgrade: len=0
* Using Stream ID: 1 (easy handle 0x55ea1034e840)
```

At this point, your client is connected via TLS. There are no issues with TLS, or the certificate.

```
> GET / HTTP/2
> Host: my.server.com
> User-Agent: curl/7.58.0
> Accept: */*
>
* Connection state changed (MAX_CONCURRENT_STREAMS updated)!
```

At this point, your client has sent a request to traefik successfully.

```
< HTTP/2 502
< content-type: text/plain; charset=utf-8
< content-length: 11
< date: Fri, 21 Dec 2018 05:23:44 GMT
<
* Connection #0 to host my.server.com left intact
Bad Gateway
```
This result is that your backend is not responding properly. Traefik cannot connect to the pods on your service defined in your ingress.

I would try curling your service and service port that you have defined in your ingress to see if it responds correctly.

**Comment by aug70** [comment]
This is trying to connect to Traefik admin UI that runs on port 8080.

**Comment by yniknafs** [comment]
@dtomcej 
Just read the thread above. I'm having same issue. My secret is not found. On curling - it doesnt seem like my client is connecting via TLS:

Below is the curl result
```
* Rebuilt URL to: https://myurl/
*   Trying IPaddr...
* TCP_NODELAY set
* Connected to myurl (IPaddr) port 443 (#0)
* ALPN, offering h2
* ALPN, offering http/1.1
* Cipher selection: ALL:!EXPORT:!EXPORT40:!EXPORT56:!aNULL:!LOW:!RC4:@STRENGTH
* successfully set certificate verify locations:
*   CAfile: /etc/ssl/cert.pem
  CApath: none
* TLSv1.2 (OUT), TLS handshake, Client hello (1):
* LibreSSL SSL_connect: SSL_ERROR_SYSCALL in connection to myurl:443
* stopped the pause stream!
* Closing connection 0
curl: (35) * LibreSSL SSL_connect: SSL_ERROR_SYSCALL in connection to myurl:443
```

Pretty stuck here. Secret exists. But ingress controller still cant find it. 

here is how i'm setting up my ingress controller:
```
apiVersion: apps/v1beta1
kind: Deployment
metadata:
  name: traefik-ingress-controller
  namespace: kube-system
  labels:
    k8s-app: traefik-ingress-lb
spec:
  replicas: 1
  selector:
    matchLabels:
      k8s-app: traefik-ingress-lb
  template:
    metadata:
      labels:
        k8s-app: traefik-ingress-lb
        name: traefik-ingress-lb
    spec:
      terminationGracePeriodSeconds: 5
      containers:
        - image: traefik
          name: traefik-ingress-lb
          imagePullPolicy: Always
          resources:
            limits:
              cpu: 200m
              memory: 30Mi
            requests:
              cpu: 100m
              memory: 20Mi
          ports:
            - containerPort: 80
            - containerPort: 8080
          args:
            - --web
            - --kubernetes
            - --logLevel=DEBUG
            - --defaultentrypoints=http,https
            - --entrypoints=Name:https Address::443 TLS
            - --entrypoints=Name:http Address::80
```

**Comment by dtomcej** [comment]
@yniknafs Your image shows that the secret `kube-system/traefik-ui-tls-cert` exists, not `default/traefik-ui-tls-cert`, as shown in your logs:

```
time="2019-02-01T21:12:16Z" level=error msg="Error configuring TLS for ingress default/ingress: secret default/traefik-ui-tls-cert does not exist"
```
Also of note, you have configured traefik to listen on port 443 for TLS via the config, but have not exposed this port as a containerport.

I am going to mark your comment as off-topic, so as not to confuse the OP in this issue.

**Comment by arthurchaloin** [comment]
I had the **exact** same problem. But it was not because of Traefik's configuration.
I solved it when I noticed that I had another reverse proxy _before_ Traefik that was decrypting the TLS requests before they were sent to my cluster. This way, **Traefik was receiving plain HTTP requests on port 443**.

However, even if this is working now, I'm still having this error:

```
Error configuring TLS for ingress default/traefik-dashboard-ingress: secret default/traefik-ui-tls-cert does not exist
```

Might be related to #4711 ?

**Comment by haf** [comment]
Trying to get this working with docker-for-desktop; same problem as above; https://stackoverflow.com/questions/55777815/traefik-not-finding-tls-secret

**Comment by cuneytcalishkan** [comment]
Hello,
I have the kubernetes ingress installed and running since January 2019 and it was all fine until today that I had to renew the certificates and did a restart on the ingress controller pods. I followed the documentation on https://docs.traefik.io/user-guide/kubernetes/#add-a-tls-certificate-to-the-ingress back in the time and it all worked well.

`Traefik version v1.7.14 built on 2019-08-14_09:46:58AM`
`GKE version 1.11.10-gke.5 `, thinking that it might be caused by outdated GKE version, I have just upgraded to version `1.13.7-gke.24 ` and still the same issue

I have tried to see the outout when the secret does not exist for real and the output is completely different, [secret-does-not-exist.txt](https://github.com/containous/traefik/files/3591214/secret-does-not-exist.txt).

For the case when the secret exists but the server is not started, [secret-not-found.txt](https://github.com/containous/traefik/files/3591215/secret-not-found.txt).

Any ideas are welcome.

P.S. I have already tried changing the order of the entry points property as `defaultentrypoints=https,http` but didn't work for me.

**Comment by jmaerki** [comment]
Similar story here: got the "does not exist" after replacing the certs. I can get/describe the certs with `--as system:serviceaccount:kube-system:traefik-ingress-controller`. In my case, interestingly, the old certs are still being presented by Traefik, even though I've killed the pods multiple times and have even made changes to the traefik daemonset. This is rather confusing since traefik doesn't get any persistent storage anywhere in the configuration. I assume "--kubernetes" places them somewhere in k8s' etcd. I've tried 1.7.9 and 17.24.
